File: src/core/GymManager.Domain/Entities/GymHouse.cs
using GymManager.Domain.Common;

namespace GymManager.Domain.Entities;

public sealed class GymHouse : AuditableEntity
{
    public string Name { get; set; } = string.Empty;
    public string Address { get; set; } = string.Empty;
    public string? Phone { get; set; }
    public string? OperatingHours { get; set; }
    public int HourlyCapacity { get; set; }
    public Guid OwnerId { get; set; }
    public User Owner { get; set; } = null!;
}

File: src/apps/gymmanager-web/src/types/gym-house.ts
export interface GymHouseDto {
  id: string;
  name: string;
  address: string;
  phone: string | null;
  operatingHours: string | null;
  hourlyCapacity: number;
  ownerId: string;
  createdAt: string;
}

export interface CreateGymHouseRequest {
  name: string;
  address: string;
  phone?: string;
  operatingHours?: string;
  hourlyCapacity: number;
}

export interface UpdateGymHouseRequest {
  name: string;
  address: string;
  phone?: string;
  operatingHours?: string;
  hourlyCapacity: number;
}
